### ISSUE TITLE
Kotlin DSL "by settings" property name limitations

### ISSUE THREAD

**Issue by karlsabo** [gradle/gradle#9631]
It's common practice to use a dot in the name of a Gradle property making it difficult to access from the settings.gradle.kts file.

```kotlin
val property.with.dot: String? by settings // Compile error
val `property.with.dot`: String? by settings // Compile error
```

workaround:
```kotlin
val propertyWithDot = settings.javaClass.getMethod("getProperty", String::class.java).invoke(settings, "property.with.dot")
```

It would be nice if settings had the "property", "findProperty", "hasProperty", and "setProperty" methods exposed on it similar to Project.

**Comment by karlsabo** [comment]
I still use properties in settings files and this would be nice to have. With configuration cache coming along I'm not sure how this plays into that. 

I use properties for disabling composite builds and adding passwords for plugin repositories. 

Anyway, outside of configuration cache I was hoping this would be easy and increases usability of Gradle.

**Comment by asinbow** [comment]
The `gradleProperties` is injected here: https://github.com/gradle/gradle/blob/v6.9.1/subprojects/core/src/main/java/org/gradle/initialization/SettingsFactory.java#L65-L66, not available when compiling, making [hasProperty() and getProperty()](https://github.com/gradle/gradle/blob/v6.9.1/subprojects/model-core/src/main/java/org/gradle/internal/extensibility/ExtensibleDynamicObject.java#L216-L224) available to Groovy but not to Kotlin unless using reflection.

**Comment by ChristianCiach** [comment]
This issue is still valid. I am trying to access dotted properties from within `settings.gradle` and cannot find a way to do so.

**Comment by little-fish** [comment]
With Kotlin DSL set as default option for new Gradle builds (Gradle 8.2), can we expect this feature to be implemented? Are there any breaking changes which prevents this to be added? Thanks.

**Comment by emilesteenkamp** [comment]
Still waiting on this issue

**Comment by karlsabo** [comment]
They closed this ticket at not planned on Sep 16, 2022. I'd recommend opening a new ticket if you're impacted and link back to this closed one for reference.
